src/linkPanel/widgets/linking.ts

```diff
@@ -445,17 +526,31 @@ namespace Private {
         attributes.forEach(attribute => {
           const option = document.createElement('option');
           option.value = attribute;
+          option.label = sharedModel.attributes[attribute].label;
           option.innerText = sharedModel.attributes[attribute].label;
           select.append(option);
         });
 
-        select.value = attributes[index] || attributes[0];
-        this._io[ioType].push(attributes[index] || attributes[0]);
-
         select.onchange = () => {
-          this.onSelectIO(ioType, index, select.value);
+          this.onSelectIO(
+            ioType,
+            index,
+            select.value,
+            (select.querySelector('option:selected') as HTMLOptionElement).label
```

Comment:
The `onchange` callback will receive a change event as its parameter where you can get the value and label of the selected item.